index.js

```diff
@@ -392,17 +379,12 @@ let Fixtures = {
      * @return {Promise} The ChakramResponse for the delete request to the server.
      */
     cleanup() {
-        if (_.isUndefined(this._headers['OAuth-Token'])) {
-            if (++this._sessionAttempt > this._maxSessionAttempts) {
-                throw new Error('Max number of login attempts exceeded!');
-            }
-
-            return this._adminLogin().then(() => this.cleanup());
+        if (this._needsLogin()) {
```

Comment:
I didn't really like the explicit checking of `OAuth-Token` here. I've moved it to its own function.

Note, however, that I didn't change the logic at all. I might have to.